airbyte-api/src/main/openapi/config.yaml

```diff
@@ -3229,6 +3231,54 @@ components:
           type: string
         stats:
           $ref: "#/components/schemas/AttemptStats"
+    AttemptFailureSummary:
+      type: object
+      required:
+        - failures
+      properties:
+        failures:
+          type: array
+          items:
+            $ref: "#/components/schemas/AttemptFailureReason"
+        partialSuccess:
+          description: True if the number of committed records for this attempt was greater than 0. False if 0 records were committed. If not set, the number of committed records is unknown.
+          type: boolean
+    AttemptFailureReason:
+      type: object
+      required:
+        - timestamp
+      properties:
+        failureOrigin:
+          $ref: "#/components/schemas/AttemptFailureOrigin"
+        failureType:
+          $ref: "#/components/schemas/AttemptFailureType"
+        externalMessage:
```

Comment:
@pmossman Sorry for the late input. Thinking a bit more about the presentation in the UI eventually, I wonder if it would make sense to allow besides the `externalMessage` also a "suggestion to fix". I guess we might have some cases where especially in `config_errors` a connector might be able to give a more clear suggestion how to solve it.

E.g. let's say we get from a GitHub repository sync a 404 repo not found. I'd imagine the `externalMessage` to be like "Repository not found". I think it would be nice if we could in a more detail view show also something like: "Please check if the repository name is valid and includes an organization as well as the used GitHub token has permissions to view this repository."

We could either have the source (of the error) create those and add a field for it here, or we can later try to map to useful hints in the frontend, in which case I think it would be useful if we'd get besides the `externalMessage` an ID-like property here, that we can really map on (like `GITHUB_SOURCE_REPO_NOT_FOUND` in this example).